### DISCUSSION TITLE
Using Kubeflow PodDefault

### DISCUSSION THREAD

**Discussion by sidineycrescencio** [elyra-ai/elyra#1249] [Q&A]
Hello,

I've created a PodDefault manifest on Kubeflow, to automatically mount a gcp credential, the problem when using Elyra is that the pods submitted via the pipeline do not have the proper labelling, therefore the secret is only mounted to the notebook server.

To get around this I can create another PodDefault using one of the labels automatically created such as "pipelines.kubeflow.org/cache_enabled=true" , but wouldn't be the most beautiful solution, and requires 2 PodDefault configs.

I wonder if there's a better way to get this working ?

Thanks in advance.

**Comment by akchinSTC** [comment]
Hi @sidineycrescencio , your workaround is probably a good one for the foreseeable future. To get this working, we would need to plumb the labels all the way down to the container/k8s api from the pipeline UI, but we don't have anything in the roadmap for this level of granularity in node configuration yet. We are working on a new version of the pipeline architecture and that may make this easier in the future. We welcome any contributions you may want to make to help with these changes ! 
A good example to check out for the changes needed for something like this would be : https://github.com/elyra-ai/elyra/pull/1203
